### PR TITLE
Add Docker support for enrollment tokens

### LINKED ISSUE #81068: Issue enrolling nodes to existing cluster on docker
In 8.0.0 nodes can enroll to an existing secured cluster by passing the `--enrollment-token <token>` parameter on node start. This approach sadly doesn't work for Docker as we don't pass arbitrary parameters passed to `docker run` to the actual entrypoint of our Docker image. The work around for now is to override the entrypoint when starting a new node that should enroll to an existing cluster, as follows for instance : 

```
docker run -it --rm -m 1g --entrypoint /usr/share/elasticsearch/bin/elasticsearch docker.elastic.co/elasticsearch/elasticsearch:8.0.0-beta1 --enrollment-token <token>
```

We should look into ways of allowing the enrollment token to be passed to docker run commands in a way that this will be relayed to the the elasticsearch executable so users don't need to override the entrypoint

### PR DESCRIPTION
**Pull request by jkakavas**
Allow users to pass an ENROLLMENT_TOKEN environment variable that
would be passed as the `--enrollment-token` parameter to the
elasticsearch executable running in the container. This allows an
elasticsearch node running in docker to enroll itself to an
existing secured cluster.

Resolves: #81068

### CODE REVIEW

File: qa/os/src/test/java/org/elasticsearch/packaging/test/EnrollmentProcessTests.java
Comment (pugnascotia) [inline qa/os/src/test/java/org/elasticsearch/packaging/test/EnrollmentProcessTests.java:85]:
```suggestion
        String createTokenResult = installation.executables().createEnrollmentToken.run("-s node").stdout;
        assertThat(createTokenResult, not(emptyOrNullString());
        final List<String> noWarningOutputLines = createTokenResult.lines()
```
Note that `.lines()` would need to be replaces with `Arrays.stream()` in `7.16`.

File: qa/os/src/test/java/org/elasticsearch/packaging/test/EnrollmentProcessTests.java
Comment (pugnascotia) [inline qa/os/src/test/java/org/elasticsearch/packaging/test/EnrollmentProcessTests.java:102]: You could use `PackagingTestCase.assertBusy()`:

```suggestion
        // Allow some time for the second node to join the cluster
        assertBusy(() ->
            // verify that the two nodes formed a cluster
            assertThat(makeRequestAsElastic("https://localhost:9200/_cluster/health", "password"), containsString("\"number_of_nodes\":2")));
```

File: docs/changelog/81082.yaml
Comment (pugnascotia) [inline docs/changelog/81082.yaml:2]: What about:

```suggestion
summary: Add Docker support for enrollment tokens
```

**Comment by pugnascotia** [comment]
Changes look good but `EnrollmentProcessTests.test20DockerAutoFormCluster` failed in CI - I'll approve when that's fixed.

File: qa/os/src/test/java/org/elasticsearch/packaging/test/EnrollmentProcessTests.java
Comment (jkakavas) [reply qa/os/src/test/java/org/elasticsearch/packaging/test/EnrollmentProcessTests.java:102]: Thanks, I hadn't seen that,will change !

File: qa/os/src/test/java/org/elasticsearch/packaging/test/EnrollmentProcessTests.java
Comment (jkakavas) [reply qa/os/src/test/java/org/elasticsearch/packaging/test/EnrollmentProcessTests.java:85]: Thanks, this is 8.0+ only so I won't backport this to 7.x

**Comment by jkakavas** [comment]
@pugnascotia this is failing on SLES workers only ( the truststore issue ) , see here: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request+packaging-tests-unix/56/. If you're happy to merge this as is you can approve, or we can wait for the underlying issue to be resolved and run CI again

**Review by pugnascotia — APPROVED** [review]
LGTM 👍 I don't think we need to wait for the SLES fix before merging 🚀

**Comment by jkakavas** [comment]
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request+packaging-tests-unix/57/ is again only SLES related, so I'll merge this

**Comment by jkakavas** [comment]
> https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request+packaging-tests-unix/57/ is again only SLES related, so I'll merge this

Actually, this depends on https://github.com/elastic/elasticsearch/pull/81077 for un-muting the tests in EnrollmentProcessTests so I'll wait for @albertzaharovits to take a look at that and then merge this.

**Pull request merged**
